release/air_tests/air_benchmarks/workloads/tune_torch_benchmark.py

```diff
@@ -0,0 +1,153 @@
+import json
+import os
+import time
+import timeit
+import click
+import numpy as np
+
+import ray
+from ray.air import ScalingConfig
+from ray.train.torch import TorchTrainer
+
+
+CONFIG = {"lr": 1e-3, "batch_size": 64, "epochs": 10}
+
+
+def prepare_mnist():
+    # Pre-download the data onto each node.
+    from benchmark_util import schedule_remote_fn_on_all_nodes
+
+    print("Preparing Torch benchmark: Downloading MNIST")
+
+    @ray.remote
+    def _download_data():
+        import torchvision
+
+        torchvision.datasets.FashionMNIST("/tmp/data_fashion_mnist", download=True)
+        return True
+
+    ray.get(schedule_remote_fn_on_all_nodes(_download_data))
+
+
+def get_trainer(num_workers: int = 4, use_gpu: bool = False):
+    """Get the trainer to be used across train and tune to ensure consistency."""
+    from torch_benchmark import train_func
+
+    def train_loop(config):
+        train_func(use_ray=True, config=config)
+
+    trainer = TorchTrainer(
+        train_loop_per_worker=train_loop,
+        train_loop_config=CONFIG,
+        scaling_config=ScalingConfig(
+            num_workers=num_workers,
+            resources_per_worker={"CPU": 2},
+            trainer_resources={"CPU": 0},
+            use_gpu=use_gpu,
+            placement_strategy="STRICT_PACK",
```

Comment:
Actually it does not consistently pass, but it seems to be closer. It could be due to constant overhead with the relatively short training time. I'll increase once more.

Reply:
With more epochs it seems to be closer, indicating a constant setup overhead: https://buildkite.com/ray-project/release-tests-pr/builds/10008

Reply:
Isn't this an odd benchmark case? Most users would presumably be using multiple GPUs per worker (as big as possible), so the packing strategy is irrelevant.

Reply:
Performance on these Data parallel systems require processes to own 1 gpu each, even with multiple gpus per node.

Reply:
It's from  the underlying allreduce algorithm

Reply:
Wait this seems like a pretty serious bug? If STRICT_PACK is feasible then PACK should always equal STRICT_PACK.

Btw we recently changed the default to SPREAD cc @matthewdeng. As @ericl pointed about before, all of our benchmarks should be using default configs, no?

Reply:
> cc @xwjiang2010 @amogkam @richardliaw @ericl STRICT_PACK makes a huge difference. Without it tune would consistently be ~60% worse than train (failing the threshold), most likely to workers being scheduled on different nodes. With it we are consistently within the allowed 20% overhead, even with different training job sizes (e.g. more epochs).
>
> Should we raise this with core (improve "PACK" placement strategy)? In any case we should document this.

Just to double check here, was this under the assumption that the default strategy was PACK and not SPREAD? (I am changing the default back to PACK and will re-run this test with default strategy)